### PR TITLE
Added user guide topic for linting your content by running Vale from the command line

### LINKED ISSUE #255: [redhat-documentation/vale-at-red-hat]: Enhance the user guide content
**Description:** As a Red Hat writer, I need information on the different ways you can use Vale 

As part of the vale-at-red-hat working user group initiative and creation of [Installing the Vale Client with the vale-at-red-hat rules](https://docs.google.com/document/d/1S9jZc2EZ8vUm33t0rrTEr6FZH0WnOXEs7ajcbQIaqv0/edit#), we have identified improvements and enhancements to the existing vale-at-red-hat [User Guide](https://redhat-documentation.github.io/vale-at-red-hat/docs/user-guide/introduction/).

**Solution:**
Migrate the following content from the [vale-at-red-hat rules](https://docs.google.com/document/d/1S9jZc2EZ8vUm33t0rrTEr6FZH0WnOXEs7ajcbQIaqv0/edit#):

- Using the vale-at-red-hat web app to check your content
- Using Vale client (with vale-at-red-hat rules) in the command line to check your content

**Additional context**
As discussed in the [vale-at-red-hat slack channel ](https://app.slack.com/client/T027F3GAJ/C0218RXJK5E)and recent open office hours workgroup to provide a CCS vale-at-red-hat onboarding toolkit.

### PR DESCRIPTION
**Pull request by michelle-purcell**
Closes #255 

As a new user of Vale and the vale-at-red-hat style, I would like detailed instructions to help me to onboard and get started.

**Audience level:** Basic

This PR migrates a section of content in this [googledoc](https://docs.google.com/document/d/1S9jZc2EZ8vUm33t0rrTEr6FZH0WnOXEs7ajcbQIaqv0/edit#heading=h.vmqjqi81msm0), which was created and tested in a collaborative effort of the vale-at-red-hat team to enhance the onboarding experience.

### CODE REVIEW

**Comment by michelle-purcell** [comment]
@themr0c / @rolfedh, 

Any ideas why the results of GitHub Acttion integrated Vale check in this PR differs from the results on the command line using the latest .vale rules?

Most of the suggestions are showing up as warnings. The only genuine warning is for 'Asciidoctor', which is a false positive that I'll submit a fix for.

Thanks
Michelle

**Comment by tmalove** [comment]
Hi @michelle-purcell  I left a few suggestions here. Let me know if you have any questions...thanks!

**Review by rolfedh — COMMENTED** [review]
Hi, Michelle. Wonderful work! Our users _really_ need this topic. I especially like the use cases!
I know this looks like a ton of feedback, but please don't be alarmed. Most of the suggestions are questions or nits based on my "reader's eye", not errors. You are welcome to adopt, discard, or improve on any of them.

File: modules/user-guide/pages/linting-vale-cli.adoc [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+====
++
+* Configured the `.vale.ini` file of your local Vale client to use the vale-at-red-hat rules, as outlined in xref:installing-vale-cli.adoc[].
+* Pulled the latest link:https://github.com/redhat-documentation/vale-at-red-hat[vale-at-red-hat] style rules from the main repository by running the `vale sync` command or by using another `git pull` sync method.
+
+.Procedure
+
+. Start a terminal session and go to the content folder containing the files that you want to analyze.
+. Enter the `vale` linter command by using the following syntax:
+[source,commandline]
```
Comment (rolfedh) [inline modules/user-guide/pages/linting-vale-cli.adoc:34]:
```suggestion
[source,terminal]
```
I'm not familiar with this source format. I've usually seen `[source,terminal]` for commands like this, and `[source,bash]` for Bash scripts.

File: modules/user-guide/pages/linting-vale-cli.adoc [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+$ vale *.* --no-wrap
+----
+[NOTE]
+====
+If you are expecting a large quantity of output in the report, use the `--no-wrap` option to make the results easier to read from the command line.
+====
++
+* To verify the style compliance of a file named `test.adoc` and remove suggestions from the output to retrieve warnings and errors, enter:
++
+[source,commandline]
```
Comment (rolfedh) [inline modules/user-guide/pages/linting-vale-cli.adoc:71]:
```suggestion
[source,terminal]
```

File: modules/user-guide/pages/linting-vale-cli.adoc [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+----
+$ vale test.adoc --minAlertLevel=warning
+----
++
+* To verify the style compliance of a file named `test.adoc` and report errors only, enter:
++
+[source,commandline]
```
Comment (rolfedh) [inline modules/user-guide/pages/linting-vale-cli.adoc:78]:
```suggestion
[source,terminal]
```

File: modules/user-guide/pages/linting-vale-cli.adoc [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+----
+$ vale test.adoc --minAlertLevel=error
+----
++
+* To verify the style compliance of a file named `test.adoc` and output errors only into a file in JSON format, enter:
++
+[source,commandline]
```
Comment (rolfedh) [inline modules/user-guide/pages/linting-vale-cli.adoc:85]:
```suggestion
[source,terminal]
```

File: modules/user-guide/pages/linting-vale-cli.adoc [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+----
+$ vale test.adoc --minAlertLevel=error --output="JSON" | tee ~/test.json
+----
++
+* To verify the style compliance of a file named `test.adoc` and output errors only into a file in JSON format, enter:
++
+* To get help including a list of valid options for running Vale, enter:
++
+[source,commandline]
```
Comment (rolfedh) [inline modules/user-guide/pages/linting-vale-cli.adoc:94]:
```suggestion
[source,terminal]
```

**Review by rolfedh — COMMENTED** [review]
I've repeated some of the suggestions, like these because some PR authors prefer having them. Other contributors find them unnecessary, redundant, unwanted, and useless. ;) 
Kindly let me know your preference.

File: modules/user-guide/pages/linting-vale-cli.adoc [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+.Prerequisites
+
+Ensure that you have completed the following prerequisite steps:
+
+* Installed the latest release link:https://vale.sh/docs/vale-cli/installation/[Vale] and link:https://docs.asciidoctor.org/asciidoctor/latest/[Asciidoctor].
++
+[IMPORTANT]
+====
+To run Vale on AsciiDoc files, Asciidoctor must be installed.
+====
```
[the remark this answers is not in the document]
Comment (michelle-purcell) [reply modules/user-guide/pages/linting-vale-cli.adoc]: This is a problem more so in the IDE, for example, in VS Code when you don't have the asciidoctor extension installed, vale doesn't work as expected.
I think you're right, all Red Hat writers will have already installed asciidoctor as part of CCS onboarding to do their daily job, so the admonition here is a bit much. I removed it. Thanks :+1:

**Comment by michelle-purcell** [comment]
> I've repeated some of the suggestions, like these because some PR authors prefer having them. Other contributors find them unnecessary, redundant, unwanted, and useless. ;-) Kindly let me know your preference.

I don't mind the in-line comments. It helps me to ensure that I catch all instances. Thanks for taking the time. It can be time-consuming, so I understand if you'd prefer it was stated only once.

File: modules/user-guide/pages/linting-vale-cli.adoc [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+:_module-type: PROCEDURE
+// End of metadata for Modular Docs
+[id="proc_linting-vale-cli_{context}"]
+= Linting your content by running Vale from the command line
+
+To analyze your content to see how it aligns with the link:https://redhat-documentation.github.io/supplementary-style-guide/[Red Hat supplementary style guide], you can run Vale from the command line. You can run Vale on individual files or an entire directory.
+
+.Prerequisites
+
+Ensure that you have completed the following prerequisite steps:
```
[the remark this answers is not in the document]
Comment (michelle-purcell) [reply modules/user-guide/pages/linting-vale-cli.adoc]: It's fluff! 
It's history!

File: modules/user-guide/pages/linting-vale-cli.adoc [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+:page-aliases: end-user-guide:running-vale-cli.adoc
+// End of metadata for Antora
+// Metadata for Modular Docs
+:context: assembly_getting-started-with-vale
+:_module-type: PROCEDURE
+// End of metadata for Modular Docs
+[id="proc_linting-vale-cli_{context}"]
+= Linting your content by running Vale from the command line
+
+To analyze your content to see how it aligns with the link:https://redhat-documentation.github.io/supplementary-style-guide/[Red Hat supplementary style guide], you can run Vale from the command line. You can run Vale on individual files or an entire directory.
```
[the remark this answers is not in the document]
Comment (michelle-purcell) [reply modules/user-guide/pages/linting-vale-cli.adoc]: :+1:  Done.

File: modules/user-guide/pages/linting-vale-cli.adoc [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+----
++
+Where:
+
+* <target> is the name of the content file that you want to analyze. Use `*` to analyze all files in the current folder.
+* <option> is an option you can use to tailor the alerts or linter output. For more information, see _Example use cases for running the Vale command_.
+* <outputfilename> is an option that you can append to the command to pipe the output to a specific file instead of at the command line.
+
+.Verification
+Vale analyzes the specified content according to the command options and the style preferences that you have specified in your `.vale.ini` configuration file.
```
[the remark this answers is not in the document]
Comment (michelle-purcell) [reply modules/user-guide/pages/linting-vale-cli.adoc]: @rolfedh - If you are completely new to vale and don't understand how it works, this statement might be helpful. You're right, it's too much here, so I moved it to the abstract, 2nd para.

File: modules/user-guide/pages/linting-vale-cli.adoc [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+By default, Vale returns a list of suggestions, warnings, and errors to the command line. For more information about the style results that are output to the command line, see xref:understanding-vale-output[].
+
+.Example use cases for running the Vale command
+
+* To verify the style compliance of a file in the current folder called `test.adoc`, enter:
```
[the remark this answers is not in the document]
Comment (michelle-purcell) [reply modules/user-guide/pages/linting-vale-cli.adoc]: Good idea :+1:

**Review by rolfedh — APPROVED** [review]
This content looks great to me! Thank you, Michelle. 
I should go have a second look at the expressions that reviewdog flagged. At a glance, the flags don't seem valid. Vale shouldn't be causing the PR checks to fail based on them.

**Pull request merged**